Person.rb

```diff
@@ -27,6 +28,11 @@ def correct_name
     @name
   end
 
+  def add_rental(rental)
+    @rentals << rental
+    rental.book = self
+  end
```

Comment:
Good effort, @Estete9! 👍 Your implementation of the `add_rental` method demonstrates an understanding of the many-to-many relationship. However, there's a small adjustment needed to fully align with the project requirements.

#### Suggested Improvement:

🔍 **Reviewing the Requirement:**
   - The requirement specifies that the method should take a `person` parameter to create the new rental.

#### Updated Method:

```ruby
def add_rental(person, date)
  # Create a new rental with the book (self) and person parameters
  Rental.new(date, self, person)
end
```

#### Explanation:
   - This adjustment ensures that the `add_rental` method conforms to the specified project requirements.
   - By including the `person` parameter in the method, you establish the 'has-many' association with the correct parameters.

Keep up the good work! If you have any questions or need further clarification, feel free to ask. 🚀

Best,
Temitope

Reply:
@topeogunleye Hi my friend, thank you for your review.
I have a quick query regarding the add_rental method you mentioned. Is this method located in the person.rb file? I noticed that if you are talking about the person.rb file, when you use self in the constructor for rental inside the person.rb, the `Rental.new(date, self, person)`, it appears to reference the person object (because we are inside the Person class). In the code you provided, it seems like we're using both self (referring to the person) and another person. Could you clarify this for me? Appreciate your insights!

Reply:
@Estete9. Sorry, This comment is meant for the book class